Algorithms_and_Data_Structures/Advanced Algorithms/Heavy_Light_Decomposition/heavy_light_decomposition.py

```
@@ -0,0 +1,514 @@
+            else:
+                self._update_chain(self.chain_head[v], v, value)
+                v = self.parent[self.chain_head[v]]
+        
+        # Update LCA
+        self.values[lca] += value
+    
+    def subtree_query(self, node: int, operation: str = 'sum') -> int:
+        """Query the subtree rooted at node."""
+        # For simplicity, we'll use a different approach
+        # In practice, you might want to maintain subtree sizes
+        result = self.values[node]
```

Ruff (0.12.2)

Local variable `result` is assigned to but never used

Remove assignment to unused variable `result`

(F841)